File: leetcode/leet412.py
# py2
# fizz-buzz

class Solution(object):
  def fizzBuzz(eslf, n):
    """
    :type n: int
    :rtype: List[str]
    """
    res = []
    for index in range(n):
      i = index + 1
      if i % 3 == 0 and i % 5 == 0:
        res.append('FizzBuzz')
      elif i % 3 == 0:
        res.append('Fizz')
      elif i % 5 == 0:
        res.append('Buzz')
      else:
        res.append(str(i))
    print res
    return res

if __name__ == '__main__':
  testData = [15]
  test = Solution()
  for item in testData:
    # print item
    test.fizzBuzz(item)

File: leetcode/leet412.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
    var res = Array.from({ length: n }, (v, index) => {
      const currentVal = index + 1
      const divisible3 = currentVal % 3 === 0
      const divisible5 = currentVal % 5 === 0
      if (divisible3 && divisible5 ) {
        return 'FizzBuzz'
      } else if (divisible3) {
        return 'Fizz'
      } else if (divisible5) {
        return 'Buzz'
      } else {
        return String(currentVal)
      }
    })
    // console.log(res);
    return res
    
};

var testData = [15]

for (let val of testData) {
  // console.log(val);
  fizzBuzz(val)
  
}